### PR TITLE
Live-1710: Multibuy 1.5

### PR DESCRIPTION
**Pull request by vitaliishpital-ledger**
Implemented Multibuy-1.5 flow

### Type

Feature

### Context

https://ledgerhq.atlassian.net/browse/LIVE-1710

### Parts of the app affected / Test plan

Exchange Buy & Sell, Portfolio, Market, Discovery

### CODE REVIEW

File: src/navigation/useDeepLinking.js [hunk 1]
```diff
@@ -36,7 +36,8 @@ function getSettingsScreen(pathname) {
 
 export function useDeepLinkHandler() {
   const { navigate } = useNavigation();
-  const { manifests } = usePlatformApp();
+  const { state } = useRemoteLiveAppContext();
+  const manifests = state.value.liveAppByIndex;
```
Comment (IAmMorrow) [inline src/navigation/useDeepLinking.js:40]: If the provided it not loaded at this time, value will be undefined. Make sure the data is loaded by checking value is defined

File: src/screens/Account/EmptyStateAccount.js [hunk 2]
```diff
@@ -8,134 +8,148 @@ import {
...
+  colors,
+}: Props) {
+  const mainAccount = getMainAccount(account, parentAccount);
+  const hasSubAccounts = Array.isArray(mainAccount.subAccounts);
+  const isToken =
+    listTokenTypesForCryptoCurrency(mainAccount.currency).length > 0;
+  const currency = getAccountCurrency(account);
+  const rampCatalog = useRampCatalog();
+
+  const canBeBought = useMemo(() => {
```
Comment (IAmMorrow) [inline src/screens/Account/EmptyStateAccount.js:43]: this function can return [boolean, boolean] or boolean, something seems fishy

File: src/screens/Exchange/Buy.js [hunk 3]
```diff
@@ -1,50 +1,124 @@
...
+  },
+};
 
-export default function Buy() {
+export default function OnRamp({ navigation, route }: Props) {
   const { t } = useTranslation();
-  const navigation = useNavigation();
   const { colors } = useTheme();
+  const rampCatalog = useRampCatalog();
+  const allCurrencies = useRampCatalogCurrencies(rampCatalog.value.onRamp);
```
Comment (IAmMorrow) [inline src/screens/Exchange/Buy.js:45]: If the provided it not loaded at this time, value will be undefined. Make sure the data is loaded by checking value is defined

File: src/screens/Exchange/Sell.js [hunk 4]
```diff
@@ -1,24 +1,100 @@
...
 const forceInset = { bottom: "always" };
 
-export default function Buy() {
-  const { colors } = useTheme();
+export default function OffRamp() {
   const { t } = useTranslation();
+  const { colors } = useTheme();
   const navigation = useNavigation();
+  const rampCatalog = useRampCatalog();
+  const allCurrencies = useRampCatalogCurrencies(rampCatalog.value.offRamp);
```
Comment (IAmMorrow) [inline src/screens/Exchange/Sell.js:33]: If the provided it not loaded at this time, value will be undefined. Make sure the data is loaded by checking value is defined

**Review by IAmMorrow — COMMENTED** [review]
Did a small pre-review, will test the PR as soon as you think it's ready :)

File: src/navigation/useDeepLinking.js (hunk 1 above)
Comment (nyurchenko-ledger) [reply src/navigation/useDeepLinking.js]: fix uploaded

File: src/screens/Account/EmptyStateAccount.js (hunk 2 above)
Comment (nyurchenko-ledger) [reply src/screens/Account/EmptyStateAccount.js:43]: missed this one, fixed

File: src/screens/Exchange/Buy.js (hunk 3 above)
Comment (nyurchenko-ledger) [reply src/screens/Exchange/Buy.js]: fix uploaded

File: src/screens/Exchange/Sell.js (hunk 4 above)
Comment (nyurchenko-ledger) [reply src/screens/Exchange/Sell.js]: fix uploaded

**Comment by Justkant** [comment]
Rebase on V3 started in https://github.com/LedgerHQ/ledger-live-mobile/tree/Multibuy-1.5-squash

**Pull request closed without merging**